### PR TITLE
L031 bug

### PR DESCRIPTION
**Pull request by NiallRees**
```
SELECT u.id, c.first_name, c.last_name, COUNT(o.user_id) FROM users as u JOIN customers as c on u.id = c.user_id JOIN orders as o on u.id = o.user_id order by o.user_id desc
```

should be fixed to

```
SELECT users.id, customers.first_name, customers.last_name, COUNT(orders.user_id) FROM users JOIN customers on users.id = customers.user_id JOIN orders on users.id = orders.user_id order by orders.user_id desc
```

But the L031 lint fix is not fixing the alias in the order by clause, so the fix currently results in

```
SELECT users.id, customers.first_name, customers.last_name, COUNT(orders.user_id) FROM users JOIN customers on users.id = customers.user_id JOIN orders on users.id = orders.user_id order by o.user_id desc
```

order by **o**.user_id desc instead of order by **orders**.user_id desc

### CODE REVIEW

**Comment by alanmcruickshank** [comment]
Thanks for adding test 🙏 . Are you planning on picking up the fix for this or do you think someone else should?

**Comment by NiallRees** [comment]
> Thanks for adding test 🙏 . Are you planning on picking up the fix for this or do you think someone else should?

Whoever gets there first! Just made this so it didn't get forgotten

**Comment by NiallRees** [comment]
Good to go

File: test/core/rules/std_test.py [hunk 1]
```diff
@@ -439,6 +439,22 @@ def assert_rule_pass_in_sql(code, sql, configs=None):
...
+            "SELECT u.id, c.first_name, c.last_name, COUNT(o.user_id) FROM users as u JOIN customers as c on u.id = c.user_id JOIN orders as o on u.id = o.user_id order by o.user_id desc",
+            "SELECT users.id, customers.first_name, customers.last_name, COUNT(orders.user_id) FROM users JOIN customers on users.id = customers.user_id JOIN orders on users.id = orders.user_id order by orders.user_id desc",
+            None,
+        ),
+        # L031 order by identifier which is the same raw as an alias but refers to a column
+        (
+            "L031",
+            "fail",
+            "SELECT u.id, c.first_name, c.last_name, COUNT(o.user_id) FROM users as u JOIN customers as c on u.id = c.user_id JOIN orders as o on u.id = o.user_id order by o desc",
+            "SELECT users.id, customers.first_name, customers.last_name, COUNT(orders.user_id) FROM users JOIN customers on users.id = customers.user_id JOIN orders on users.id = orders.user_id order by o desc",
```
Comment (pwildenhain) [inline test/core/rules/std_test.py:455]: Oh wow, really nice thinking ahead here 🧠 ✔️ with not replace straight column references

**Review by pwildenhain — APPROVED** [review]
🐛 👢 🙌

File: test/core/rules/std_test.py (hunk 1 above)
Comment (NiallRees) [reply test/core/rules/std_test.py:455]: Cheers 🍻

**Pull request merged**